drivers/sensor/tmag5273/Kconfig

```diff
@@ -0,0 +1,364 @@
+		select TMAG5273_AXIS_DEPENDENT_CALC_EN
+		help
+		  X 1st, Z 2nd
+
+	config TMAG5273_AXIS_DEPENDENT_CALC_RUNTIME
+		bool "@ runtime"
+		select TMAG5273_AXIS_DEPENDENT_CALC_EN
+		help
+		  Set option during runtime.
+		  Default: none
+endchoice
+
+choice TMAG5273_MEAS_RANGE_XYZ_CHOICE
+		prompt "magnetic-field XY Zmeasurement range"
+		default TMAG5273_XYZ_MEAS_RANGE_HIGH
+		depends on TMAG5273_MAGNETIC_EN
+		help
+		  XYZ-measurement range for magnetic-field values.
+
+		config TMAG5273_XYZ_MEAS_RANGE_LOW
+			bool "+/-40mT (TMAG5273A1) or +/-133mT (TMAG5273A2)"
+
+		config TMAG5273_XYZ_MEAS_RANGE_HIGH
+			bool "+/-80mT (TMAG5273A1) or +/-266mT (TMAG5273A2)"
+
+		config TMAG5273_XYZ_MEAS_RANGE_RUNTIME
+			bool "select at runtime"
+			help
+			  Select measurement range at runtime.
+			  The sensor will startup using the measurement range "high".
+endchoice
+
+choice TMAG5273_MAG_GAIN_CORRECTION_CHANNEL_CHOICE
+	prompt "axis for magnitude gain correction"
+	default TMAG5273_MAG_GAIN_CORRECTION_CH_1
+
+	config TMAG5273_MAG_GAIN_CORRECTION_CH_1
+		bool "1st channel"
+	config TMAG5273_MAG_GAIN_CORRECTION_CH_2
+		bool "2nd channel"
+endchoice
+
+comment "device Configuration"
+
+config TMAG5273_NEED_INT_PIN
+	bool
+
+choice TMAG5273_OPERATING_MODE_CHOICE
+	prompt "'Active' operation mode"
+	default TMAG5273_OPERATING_MODE_CONTINUOUS
+	help
+	  Specifies the sensor mode which is used if set to "active".
+
+	config TMAG5273_OPERATING_MODE_CONTINUOUS
+		bool "continuous"
+
+	config TMAG5273_OPERATING_MODE_STANDBY
+		bool "standby"
+endchoice
+
+choice TMGAG5273_TRIGGER_MODE_CHOICE
+	prompt "trigger through"
+	default TMAG5273_TRIGGER_MODE_I2C
+
+	config TMAG5273_TRIGGER_MODE_I2C
+		bool "i2C command"
+	config TMAG5273_TRIGGER_MODE_INT
+		bool "through INT pin"
+		depends on GPIO
+		select TMAG5273_NEED_INT_PIN
+endchoice
+
+if PM_DEVICE
+
+	choice TMAG5273_MODE_SUSPENDED_CHOICE
+		prompt "'Suspend' operation mode"
+		default TMAG5273_MODE_SUSPENDED_SLEEP
+		depends on PM_DEVICE
+		help
+		  Specifies the sensor mode which is set on suspension command.
+
+		config TMAG5273_MODE_SUSPENDED_SLEEP
+			bool "suspend to sleep"
+
+		config TMAG5273_MODE_SUSPENDED_WAKEUP_AND_SLEEP
+			bool "suspend to 'wake-up and sleep'"
+	endchoice
+
+	config TMAG5273_I2C_VCC_SLEW_RATE_SEQUENCE
+		bool "use Vcc ramp < 3V/ms workaround"
+		depends on TMAG5273_TRIGGER_MODE_I2C
+		help
+		  As stated in the datasheet, if the VCC ramp rate is slower than 3V/ms,
+		  the device needs to run a wake-up and sleep cycle after power-up or
+		  power-up reset to avoid I2C address glitch during sleep mode.
+		  If the sensor is randomly not reachable after power-up, activate this option.
+
+	choice TMAG5273_SLEEPTIME_CHOICE
+		prompt "sleeptime"
+		default TMAG5273_SLEEPTIME_1000MS
+		depends on TMAG5273_MODE_SUSPENDED_WAKEUP_AND_SLEEP
+		help
+		  Selects the time spent in low power mode between conversions.
+
+		config TMAG5273_SLEEPTIME_1MS
```

Comment:
It would probably make sense to use an int option with a range here instead of a choice and simply default it to 1000 ms

Reply:
This does have it's charme, however, there are a lot of unsupported values in between which needs to be handled by either
- raising errors during initialization
- round to the "nearest" value (though the user may no know what the exact time is).

The configuration variable is using the same "philosphy" as after other configurations like averaging etc. which usually has a power of 2 scaling.

But granted, e.g. the measurement range is behaving a more ranged-based way for runtime changes (though then, it is implemented as an attribute rather than an configuration option).

Reply:
is now implemented as an enum in the dts-file